lib/awful/hotkeys_popup/widget.lua

```diff
@@ -37,6 +37,29 @@
 --
 --    hotkeys_popup.add_hotkeys(firefox_keys)
 --
+--
+-- Example of having different types of hotkey popups:
+--
+--    local globalkeys = awful.util.table.join(
+--        awful.key({modkey}, "/", function()
+--          hotkeys_popup.show_help()
+--        end, nil, {
+--          description = "show help (all)", group="HELP"
+--        }),
+--        awful.key({"Shift", modkey}, "/", function()
+--          hotkeys_popup.show_help(nil, nil, {show_awesome_keys=false})
+--        end, nil, {
+--          description = "show help for current app", group="HELP"
+--        }),
+--        awful.key({altkey, modkey}, "/", function()
+--          hotkeys_popup.show_help({}, nil, {show_awesome_keys=true})
+--        end, nil, {
+--          description = "show help for awesome only", group="HELP"
+--        })
+--        -- (more hotkeys goes here)
+--    )
```

Comment:
Maybe you should move these code examples to the examples. So that it gets syntax checking and is run at every CI build to check for breakage.

Reply:
that wouldn't run any code, as those are callbacks

Reply:
You can run those callbacks by calling `trigger` on the key object.
You'd just have to hide the calls and setup for it to keep the visible portion of the example the same.

Reply:
feel free to add it if you want